pkg/domain/entities/account.go

```diff
@@ -41,7 +42,7 @@ func NewAccount(name, cpf, secret string, balance int) (Account, error) {
 
 	hash, err := HashGenerator(secret)
 	if err != nil {
-		return Account{}, err
+		return Account{}, fmt.Errorf("%s: %w", ErrToGenerateHash, err)
```

Comment:
it seems to me that you're interested more in ErrToGenerateHash as value than the error returned by bcrypt package, won't you? If so, wouldn't be better to wrap it like:
```suggestion
		return Account{}, fmt.Errorf("%w: %s", ErrToGenerateHash, err)
```